lib/media/track/localaudiotrack.js

```diff
@@ -20,14 +22,114 @@ const LocalMediaAudioTrack = mixinLocalMediaTrack(AudioTrack);
  * @emits LocalAudioTrack#started
  * @emits LocalAudioTrack#stopped
  */
+
 class LocalAudioTrack extends LocalMediaAudioTrack {
   /**
    * Construct a {@link LocalAudioTrack} from a MediaStreamTrack.
    * @param {MediaStreamTrack} mediaStreamTrack - An audio MediaStreamTrack
    * @param {LocalTrackOptions} [options] - {@link LocalTrack} options
    */
   constructor(mediaStreamTrack, options) {
+    // Note(mpatwardhan): by default workaround for WebKitBug1208516 will be enabled on Safari browsers
+    // although the bug is seen  mainly on iOS devices, we do not have a reliable way to tell iOS from MacOs
+    // userAgent on iOS pretends its macOs if Safari is set to request desktop pages.
+    const workaroundWebKitBug1208516 = guessBrowser() === 'safari';
+
+    options = Object.assign({
+      workaroundWebKitBug1208516,
+      getUserMedia,
+    }, options);
+
     super(mediaStreamTrack, options);
```

Comment:
is it okay to update the id for a LocalMediaTrack ? I am bit concerned about changing the id.  updating isEnabled/isStoppped makes sense though. .

Reply:
@makarandp0 We should make sure we honor the existing rule that `LocalMediaTrack.id === LocalMediaTrack.mediaStreamTrack.id`. Otherwise it can be considered as a breaking change.

Reply:
@manjeshbhargav  I think we need to honor the contract that the id of the LocalTrack does not change once created. id is a only unique field ( since sid is not available yet) that developers would  to associate the LocalTrack with dom element - if we changed the id they won't be able to associate it anymore and that would  break the applications. 
I suspect it's pretty common pattern.

Reply:
@makarandp0 I want to do the other way around.
```js
Object.defineProperties(mediaStreamTrack, {
  id: {
    enumerable: true,
    get() {
      return localMediaTrack.id;
    }
  }
});
```

Reply:
I am worried about this lets talk more today.

Reply:
As we talked offline, decided to update the trackId that we sent in SDPs instead of changing id of the MediaStreamTrack